azure/table_azure_eventgrid_domain.go

```diff
@@ -0,0 +1,402 @@
+			},
+			{
+				Name:        "private_endpoint_connections",
+				Description: "List of private endpoint connections.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("PrivateEndpointConnections"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ID").Transform(idToAkas),
+			},
+
+			// Azure standard column
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Location").Transform(formatRegion).Transform(toLower),
+			},
+			{
+				Name:        "resource_group",
+				Description: ColumnDescriptionResourceGroup,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(extractResourceGroupFromID),
+			},
+			{
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+type DomainInfo struct {
+	Identity                             interface{}
+	SystemData                           interface{}
+	PrivateEndpointConnections           interface{}
+	ProvisioningState                    *string
+	Endpoint                             *string
+	InputSchema                          interface{}
+	InputSchemaMapping                   interface{}
+	MetricResourceID                     *string
+	PublicNetworkAccess                  interface{}
+	InboundIPRules                       interface{}
+	DisableLocalAuth                     *bool
+	AutoCreateTopicWithFirstSubscription *bool
+	AutoDeleteTopicWithLastSubscription  *bool
+	Sku                                  interface{}
+	Location                             *string
+	Tags                                 map[string]*string
+	ID                                   *string
+	Name                                 *string
+	Type                                 *string
+}
+
+//// LIST FUNCTION
+
+func listEventGridDomains(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listEventGridDomains")
+
+	// Create session
+	session, err := GetNewSession(ctx, d, "MANAGEMENT")
+	if err != nil {
+		return nil, err
+	}
+
+	subscriptionID := session.SubscriptionID
+	client := eventgrid.NewDomainsClient(subscriptionID)
+	client.Authorizer = session.Authorizer
+	top := int32(100)
+
+	result, err := client.ListBySubscription(ctx, "", &top)
```

Comment:
check with client.ListBySubscription(ctx, "", nil)

Reply:
It will work if we specify top param as nil, it will fallback to default value that is 20 per page